cmd/checkpoint/main.go

```diff
@@ -311,6 +334,28 @@ func getLocalParentPods() (map[string]*v1.Pod, error) {
 	return podListToParentPods(&podList), nil
 }
 
+// getLocalRunningPods uses the /runningpods/ kubelet api to retrieve the local container runtime pod state
+func getLocalRunningPods() (map[string]*v1.Pod, error) {
+	// TODO(aaron): The kubelet api is currently secured by a self-signed cert. We should update this to actually verify at some point
+	client := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
+	resp, err := client.Get(kubeletAPIRunningPodsURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to contact kubelet pod api: %v", err)
+	}
+
+	defer resp.Body.Close()
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response: %v", err)
+	}
+
+	var podList v1.PodList
+	if err := json.Unmarshal(b, &podList); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal podlist: %v", err)
+	}
+	return filterPodList(&podList, func(*v1.Pod) bool { return true }), nil
```

Comment:
This never filters anything, is this code just reliant on a side effect of `filterPodList`?